### PR TITLE
Minor improvements

### PR DESCRIPTION
**Pull request by Jeeyb**

### CODE REVIEW

File: types/types.go [hunk 1]
```diff
@@ -15,6 +15,10 @@ func (id ID) String() string {
 	return hex.EncodeToString(id[:])
 }
 
+func (id ID) IsEmpty() bool {
+	return id == [32]byte{}
+}
+
```
Comment (offerm) [inline types/types.go:21]: is this valid?

File: types/types.go [hunk 2]
```diff
@@ -58,15 +62,15 @@ func (lt LoadedTxs) Load(hash string, loader ID) {
...
 	lt.lock.Lock()
 	defer lt.lock.Unlock()
 	tx, ok := lt.txs[hash]
 	if !ok {
-		return nil
+		return ID{}
 	}
 	delete(lt.txs, hash)
-	return &tx
+	return tx.GetLoader()
```
Comment (offerm) [inline types/types.go:73]: please add a test:

add a transaction and remove make sure you get the expected loader.
remove a non existing transaction, make sure you get empty ID.

File: types/types.go (hunk 1 above)
Comment (Jeeyb) [reply types/types.go:21]: Yes

**Review by offerm — APPROVED** [review]
LGTM

**Review by leonbiton1 — APPROVED** [review]

**Pull request merged**